canonical-web-and-design/maas-ui
# feat(ui): add UI support for VMFS7

## Done

- Added UI support for VMFS7 storage
- Converted storage settings form to TypeScript

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a machine with a boot disk smaller than 32GB
- Click "Change storage layout" and choose "VMFS7"
- Check that an error notification displays
- Go to the storage tab of a machine with a boot disk larger than 32GB and again choose "VMFS7"
- The machine should successfully change layout and the the datastores/available/used storage tables should show the correct data. Note that the fstype is incorrectly showing up as `vmfs6`, as mentioned [here](https://github.com/canonical-web-and-design/maas-ui/issues/2772#issuecomment-904307219)
- Go to /MAAS/r/settings/storage and change the storage layout to "VMFS7"
- Check that a warning shows up below

## Fixes

Fixes #2772 

## Launchpad issue

[lp#1931736](https://bugs.launchpad.net/bugs/1931736)
